### PR TITLE
virtio_serial: accept specified chardev_path and serial_name

### PR DESCRIPTION
**Pull request by yanan-fu**
Some test need use specified chardev_path and serial_name for
testing, make it accept these definiation.
If it is not defined, use the default value that generated by
the test framework.

id: 1786555
Signed-off-by: Yanan Fu <yfu@redhat.com>

### CODE REVIEW

**Comment by yanan-fu** [comment]
@xutian could you help review, thanks!

**Comment by yanan-fu** [comment]
@xutian After add `chardev_path`, we can but I do not want to change the interface of `serials_define_by_params`, `chardev_define_by_params and `.  So i pass 'serial_filename' again.
I want to reduce the effect for other testing.

**Comment by yanan-fu** [comment]
@xutian After add `chardev_path`, we can but I do not want to change the interface of `serials_define_by_params`, `chardev_define_by_params and `.  So i pass 'serial_filename' again.
I want to reduce the effect for other testing.

File: shared/cfg/base.cfg [hunk 1]
```diff
@@ -146,7 +146,9 @@ q35, arm64-pci, arm64-mmio:
 # chardev_host = $host_ip (127.0.0.1 by default)
 # chardev_ipv4 = yes
 # chardev_ipv6 = yes
+# chardev_path = *** 'path' for chardev, if not define, framework have default
```
Comment (xutian) [inline shared/cfg/base.cfg]: better to move comments in your code.

File: shared/cfg/base.cfg [hunk 2]
```diff
@@ -146,7 +146,9 @@ q35, arm64-pci, arm64-mmio:
...
 # chardev_name = vdagent/smartcard/usbredir (only for spicevmc and spiceport)
+# serial_name = *** 'name' for serial device, if not define, use serial id
```
Comment (xutian) [inline shared/cfg/base.cfg]: too.

File: shared/cfg/base.cfg (hunk 1 above)
Comment (xutian) [reply shared/cfg/base.cfg]: configuration for some special case, so suggest to leave comments in code don't introduce it in base.cfg

Comment (yanan-fu) [reply shared/cfg/base.cfg]: No strong idea about this, it is just a comment. remove it now. thanks!

File: shared/cfg/base.cfg (hunk 2 above)
Comment (yanan-fu) [reply shared/cfg/base.cfg]: same, remove this comment.

**Comment by xutian** [comment]
LGTM

**Comment by yanan-fu** [comment]
@nanliu-r could you help review, thanks!

**Comment by nanliu-r** [comment]
> @nanliu-r could you help review, thanks!

Test PASS and can get the right qemu command, thanks.

(1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket: PASS (57.77 s)

ACK.

**Pull request merged**